src/status_im/hardwallet/core.cljs

```diff
@@ -81,14 +112,167 @@
+
+(fx/defn generate-mnemonic
+  [cofx]
+  (let [{:keys [pairing]} (get-in cofx [:db :hardwallet :secrets])]
+    {:hardwallet/generate-mnemonic {:pairing pairing}}))
+
+(fx/defn on-card-connected
+  [{:keys [db] :as cofx} data]
+  (log/debug "[hardwallet] card connected " data)
+  (let [step (get-in db [:hardwallet :return-to-step] :begin)]
+    (fx/merge cofx
+              {:db                              (-> db
+                                                    (assoc-in [:hardwallet :setup-step] step)
+                                                    (assoc-in [:hardwallet :card-connected?] true))
+               :hardwallet/get-application-info nil}
+              (navigation/navigate-to-cofx :hardwallet-setup nil))))
+
+(fx/defn on-card-disconnected
+  [{:keys [db]} _]
+  (log/debug "[hardwallet] card disconnected ")
+  {:db (assoc-in db [:hardwallet :card-connected?] false)})
+
+(fx/defn load-preparing-screen
+  [{:keys [db]}]
+  {:db       (assoc-in db [:hardwallet :setup-step] :preparing)
+   :dispatch ^:flush-dom [:hardwallet/install-applet-and-init-card]})
+
+(fx/defn install-applet-and-init-card
+  [{:keys [db]}]
+  {:hardwallet/install-applet-and-init-card nil})
+
+(fx/defn on-install-applet-and-init-card-success
+  [{:keys [db]} secrets]
+  (let [secrets' (js->clj secrets :keywordize-keys true)]
+    {:db (-> db
+             (assoc-in [:hardwallet :setup-step] :secret-keys)
+             (assoc-in [:hardwallet :secrets] secrets'))}))
+
+(fx/defn on-install-applet-and-init-card-error
+  [{:keys [db]} error]
+  (log/debug "[hardwallet] install applet and init card error: " error)
+  {:db (-> db
+           (assoc-in [:hardwallet :setup-step] :error)
+           (assoc-in [:hardwallet :return-to-step] :begin)
+           (assoc-in [:hardwallet :setup-error] error))})
+
+(fx/defn on-pairing-success
+  [{:keys [db]} pairing]
+  {:db (-> db
+           (assoc-in [:hardwallet :setup-step] :card-ready)
+           (assoc-in [:hardwallet :secrets :pairing] pairing))})
+
+(fx/defn on-pairing-error
+  [{:keys [db]} error]
+  (log/debug "[hardwallet] pairing error: " error)
+  {:db (-> db
+           (assoc-in [:hardwallet :setup-step] :error)
+           (assoc-in [:hardwallet :return-to-step] :secret-keys)
+           (assoc-in [:hardwallet :setup-error] error))})
+
+(fx/defn on-generate-mnemonic-success
+  [{:keys [db]} mnemonic]
+  {:db (-> db
+           (assoc-in [:hardwallet :setup-step] :recovery-phrase)
+           (assoc-in [:hardwallet :secrets :mnemonic] mnemonic))})
+
+(fx/defn on-generate-mnemonic-error
+  [{:keys [db]} error]
+  (log/debug "[hardwallet] generate mnemonic error: " error)
+  {:db (-> db
+           (assoc-in [:hardwallet :setup-step] :error)
+           (assoc-in [:hardwallet :return-to-step] :card-ready)
+           (assoc-in [:hardwallet :setup-error] error))})
+
+(fx/defn on-pin-validated [{:keys [db] :as cofx}]
+  (let [pin (get-in db [:hardwallet :pin :original])]
+    (fx/merge cofx)))
```

Comment:
?

Reply:
@flexsurfer Removed this code as we don't allow set PIN for now